swift-mode.el

```diff
@@ -693,6 +693,175 @@ We try to constraint those lookups by reasonable number of lines.")
                "warning: " (message) line-end))
      :modes swift-mode))
 
+;;; BUILDING
+
+(defun swift-mode-find-workspaces-for-directory(directory)
+  (mapcar 'string-trim (split-string (shell-command-to-string
+                                      (concat
+                                       (concat "find " (substring directory 0 -1))
+                                       (concat " -name \"*workspace\" -maxdepth 1"))))))
+
+(defun swift-mode-find-projects-for-directory(directory)
+  (mapcar 'string-trim (split-string (shell-command-to-string
+                                      (concat
+                                       (concat "find " (substring directory 0 -1))
+                                       (concat " -name \"*xcodeproj\" -maxdepth 1"))))))
+
+(defun swift-mode-find-schemes-for-workspace(workspace)
+  (mapcar 'string-trim
+          (cdr (cdr
+                (split-string (shell-command-to-string
+                               (concat
+                                (concat "xcodebuild -workspace " workspace) " -list")) "\n")))))
+
+(defun swift-mode-kill-compilation-if-exists()
```

Comment:
It seems to fix the never ending xcodebuild's. Maybe there is something else that could fix this but killing the `*compilation*` buffer fixes the issue.